extensions/Worker.Extensions.ServiceBus/src/ServiceBusSessionMessageActionsConverter.cs

```diff
@@ -49,7 +54,17 @@ public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
                     throw new InvalidOperationException("Expecting SessionLockedUntil within binding data of session actions and value was not present.");
                 }
 
-                var sessionActionResult = new ServiceBusSessionMessageActions(_settlement, sessionId!.ToString(), sessionLockedUntil.GetDateTimeOffset());
+                // Logic for if isBatched is true, then sessionIdArray will be used to get the sessionId.
+                var sessionIdRepeatedFieldArray = sessionIdArray as RepeatedField<string>;
```

Comment:
`RepeatedField<T>` implements `IList<T>`, so I recommend casting to that as it will be less dependent on precise payload (and more testable!)

```suggestion
                var sessionIdRepeatedFieldArray = sessionIdArray as IList<string>;
```